test/tap/tests/unit/genai_config_query_unit-t.cpp

```
@@ -0,0 +1,144 @@
+
+	test_init_minimal();
+
+	{
+		AdminFixture fixture;
+		Config_Tool_Handler handler(nullptr);
+
+		const json tool_list = handler.get_tool_list();
+		bool found_query = false;
+		for (const auto& tool : tool_list["tools"]) {
+			if (tool.value("name", "") == "query") {
+				found_query = true;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)